pkg/sqlcache/informer/synthetic_watcher.go

```diff
@@ -0,0 +1,139 @@
+package informer
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/dynamic"
+)
+
+type SyntheticWatcher struct {
+	resultChan   chan watch.Event
+	stopChan     chan struct{}
+	doneChan     chan struct{}
+	stopChanLock sync.Mutex
+}
+
+func newSyntheticWatcher() *SyntheticWatcher {
+	return &SyntheticWatcher{
+		stopChan:   make(chan struct{}),
+		doneChan:   make(chan struct{}),
+		resultChan: make(chan watch.Event, 0),
+	}
+}
+
+func (rw *SyntheticWatcher) watch(client dynamic.ResourceInterface, options metav1.ListOptions, interval time.Duration) (*SyntheticWatcher, error) {
```

Comment:
It'd be nice if we could make this code context-aware, in order for it to shut down properly in case Steve is shutting down.

In practice this means having a context parameter here and passing it down the forever-loop goroutine.

Reply:
I think this should be part of a separate PR.  

First, wouldn't we want to pass the context to `NewInformer` and then pass it to all of `client.Watch()` in the default `WatchFunc` and `client.List()` in the `ListFunc` as well as the synthetic `WatchFunc`?  I'll add it for now, but I would expect contexts are more meaningful if the invoker has access to them.